### PR TITLE
feat: #849 adjust styling manage permissions

### PR DESCRIPTION
**Pull request by J0taFerreira**
- updated all dashboard names and references to manage permissions
- adjusted  table style
- adjusted dropdown styles
- added overlay on the table when no application is selected

notes:
- some of the styling is on the common repo
- did not implemented jump to page functionality due to primevue component limitation
- the icons on the paginator's prev and next button did not got changed due to primevue component limitation

### CODE REVIEW

**Comment by MCatherine1994** [comment]
Hi Jota, this table header part doesn't look like the same style as in the design, for example, the background color will be white. 

Screen Shot 2023-09-14 at 2 42 14 PM

Screen Shot 2023-09-14 at 2 42 51 PM

**Comment by MCatherine1994** [comment]
Other than that, I'm just a little bit worried about to show the "edit" icon, cause it won't do anything, might confused the user a bit. Shall we hide it for now maybe? How you feel @ianliuwk1019 ?

image

**Comment by MCatherine1994** [comment]
The background when show the table should also be full height?

image

**Comment by MCatherine1994** [comment]
the rest looks good to me, thanks Jota!!

File: frontend/src/views/DashboardView.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <script setup lang="ts">
-import Dashboard from '@/components/dashboard/Dashboard.vue';
+import ManagePermissions from '@/components/managePermissions/ManagePermissions.vue';
 </script>
 
 <template>
-    <Dashboard />
+    <ManagePermissions />
```
Comment (ianliuwk1019) [inline frontend/src/views/DashboardView.vue:6]: Wondering why you still keep "DashboardView.vue" while it seem entirely been replaced by ManagePermissions.vue?
I am not aware we are anticipating some other components or extra styling or logic here, or will we?

**Comment by ianliuwk1019** [comment]
The wording on sideNav is not consistent with design:
image

**Review by ianliuwk1019 — CHANGES_REQUESTED** [review]

**Comment by ianliuwk1019** [comment]
> Other than that, I'm just a little bit worried about to show the "edit" icon, cause it won't do anything, might confused the user a bit. Shall we hide it for now maybe? How you feel @ianliuwk1019 ?
> 
> image

Agree to hide it.

**Comment by ianliuwk1019** [comment]
This wording does not match Figma also (I am not sure if I am looking into one different than yours?)
image

File: frontend/src/components/managePermissions/ManagePermissionsTitle.vue [hunk 2]
```diff
@@ -14,7 +14,10 @@ const props = defineProps({
 <template>
     <div class="row">
         <div class="col-6">
-            <PageTitle title="Dashboard" subtitle="Manage permissions" />
+            <PageTitle
+                title="Manage permissions"
+                subtitle="Add permissions or edit users and delegated admins for the selected application"
+            />
```
[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/ManagePermissionsTitle.vue:20]: Hello! I'll take a look. Thanks!

File: frontend/src/views/DashboardView.vue (hunk 1 above)
Comment (J0taFerreira) [reply frontend/src/views/DashboardView.vue:6]: I think I missed this one. Thanks!

**Comment by J0taFerreira** [comment]
> Hi Jota, this table header part doesn't look like the same style as in the design, for example, the background color will be white.
> 
> Screen Shot 2023-09-14 at 2 42 14 PM Screen Shot 2023-09-14 at 2 42 51 PM

most of the table styling is on the common repo file, but I'll take a second look

File: frontend/src/components/managePermissions/ManagePermissionsTitle.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/ManagePermissionsTitle.vue:20]: Screenshot from 2023-09-15 09-32-42

I believe it's correct.

File: frontend/src/components/managePermissions/ManagePermissions.vue [hunk 3]
```diff
@@ -85,17 +85,17 @@ async function deleteUserRoleAssignment(
...
         <div class="application-group">
-            <label>Select an application you would like to grant access</label>
+            <label>You are modifying access in this application:</label>
             <Dropdown
                 v-model="selectedApplication"
                 @change="selectApplication"
                 :options="selectApplicationOptions"
                 optionLabel="application_description"
-                placeholder="Choose an option"
+                placeholder="Choose or enter an application to manage permissions"
```
[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/ManagePermissions.vue:98]: Agree, it seens odd "enter" on a dropdown

File: frontend/src/components/managePermissions/ManagePermissions.vue [hunk 4]
```diff
@@ -28,10 +28,10 @@ onMounted(async () => {
...
     applicationsUserAdministers.value = (
         await applicationsApi.getApplications()
-    ).data;
-    if (isApplicationSelected) {
-        await getAppUserRoleAssignment();
-    }
+        ).data;
+        if (isApplicationSelected) {
+            await getAppUserRoleAssignment();
+        }
```
[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/ManagePermissions.vue:34]: Just fixed. Thanks!

**Comment by J0taFerreira** [comment]
> This wording does not match Figma also (I am not sure if I am looking into one different than yours?) image

Do you mean the position of the words or the text itself?

File: frontend/src/components/managePermissions/ManagePermissionsTitle.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (ianliuwk1019) [reply frontend/src/components/managePermissions/ManagePermissionsTitle.vue:20]: Thanks for removing the "delegated admin" wording for now.
I checked again, but why when empty list, the wording is now on the third line (even on full screen)?
image

File: frontend/src/components/managePermissions/ManagePermissions.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (ianliuwk1019) [reply frontend/src/components/managePermissions/ManagePermissions.vue:98]: Did you adjust this yet?

[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/ManagePermissions.vue]: I thought I did, sorry! Just pushed the change

**Comment by MCatherine1994** [comment]
> > Hi Jota, this table header part doesn't look like the same style as in the design, for example, the background color will be white.
> > Screen Shot 2023-09-14 at 2 42 14 PM Screen Shot 2023-09-14 at 2 42 51 PM
> 
> most of the table styling is on the common repo file, but I'll take a second look

Hi Jota, as we discussed before, the table header is just for FAM I believe? 
image

File: frontend/src/components/managePermissions/UserDataTable.vue [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+            <p>
+                Are you sure you want to remove
+                <strong>{{ confirmDeleteData.role }}</strong> access to
+                <strong>{{ confirmDeleteData.userName }}</strong> in
+                <strong>{{ selectedApplicationDisplayText }}</strong>
+            </p>
+        </template>
+    </ConfirmDialog>
+    <div class="data-table-container">
+        <span class="custom-data-table-bg-layer"></span>
```
Comment (ianliuwk1019) [inline frontend/src/components/managePermissions/UserDataTable.vue:95]: Hi Jota,
It looks strange to me this line.
I am curious what this <span> does when it is just by itself?
(`<span class="custom-data-table-bg-layer"></span>`)

**Review by ianliuwk1019 — CHANGES_REQUESTED** [review]

File: frontend/src/components/managePermissions/UserDataTable.vue (hunk 5 above)
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/UserDataTable.vue:95]: For the table to align with the body automatic was easier to make the background layer a separate component.

File: frontend/src/components/managePermissions/ManagePermissionsTitle.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (ianliuwk1019) [reply frontend/src/components/managePermissions/ManagePermissionsTitle.vue:20]: I see this is fixed.

[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/ManagePermissionsTitle.vue:20]: Yes it is.

**Comment by MCatherine1994** [comment]
Hi Jota, I already published a new version for the common repo, so let's update the dependency version for "@bcgov-nr/nr-fsa-theme" to the latest v1.2.0. So change to `"@bcgov-nr/nr-fsa-theme": "^1.2.0",`

Screen Shot 2023-09-18 at 3 48 27 PM

**Comment by OlgaLiber2** [comment]
@J0taFerreira please take out the "edit" icon from the action column. We will add it when we have the functionality built. TY @MCatherine1994

**Review by MCatherine1994 — APPROVED** [review]

**Comment by MCatherine1994** [comment]
Screen Shot 2023-09-19 at 10 37 29 AM
Just notice a new thing, the background color is responsive for the window size, but the table height is not. And it looks weird in this case.

**Review by ianliuwk1019 — APPROVED** [review]

**Pull request merged**